tests/v2/validation/registries/registry_test.go

```diff
@@ -0,0 +1,221 @@
+	cnis := clustersConfig.CNIs
+	nodesAndRoles := clustersConfig.NodesAndRolesRKE1
+
+	client, err := rancher.NewClient(token, session)
+	require.NoError(rt.T(), err)
+
+	err = ranchersetup.PostRancherInstall(client, adminPassword)
+	require.NoError(rt.T(), err)
+
+	corralConfig = corral.CorralConfigurations()
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(rt.T(), err)
+
+	configPackage = corral.CorralPackagesConfig()
+
+	path = configPackage.CorralPackageImages[corralAuthDisabledImageName]
+	_, err = corral.CreateCorral(testSession, corralAuthDisabledName, path, true, configPackage.Cleanup)
+	require.NoError(rt.T(), err)
+
+	registryDisabledFqdn, err := corral.GetCorralEnvVar(corralAuthDisabledName, "registry_fqdn")
+	require.NoError(rt.T(), err)
+
+	var privateRegistriesNoAuth []management.PrivateRegistry
+	privateRegistry := management.PrivateRegistry{}
+	privateRegistry.URL = registryDisabledFqdn
+	privateRegistry.IsDefault = true
+	privateRegistry.Password = ""
+	privateRegistry.User = ""
+	privateRegistriesNoAuth = append(privateRegistriesNoAuth, privateRegistry)
+
+	noAuthClusterName, err := ranchersetup.CreateTestCluster(client, client, 1, "noauthregcluster", cnis[0], kubernetesVersions[0], nodesAndRoles, privateRegistriesNoAuth)
+	require.NoError(rt.T(), err)
+
+	corralConfig = corral.CorralConfigurations()
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(rt.T(), err)
+
+	configPackage = corral.CorralPackagesConfig()
+
+	path = configPackage.CorralPackageImages[corralAuthEnabledImageName]
+	_, err = corral.CreateCorral(testSession, corralAuthEnabledName, path, true, configPackage.Cleanup)
+	require.NoError(rt.T(), err)
+
+	registryEnabledUsername, err := corral.GetCorralEnvVar(corralAuthEnabledName, "registry_username")
+	require.NoError(rt.T(), err)
+	registryEnabledPassword, err := corral.GetCorralEnvVar(corralAuthEnabledName, "registry_password")
+	require.NoError(rt.T(), err)
+	registryEnabledFqdn, err := corral.GetCorralEnvVar(corralAuthEnabledName, "registry_fqdn")
+	require.NoError(rt.T(), err)
+
+	var privateRegistriesAuth []management.PrivateRegistry
+	privateRegistry = management.PrivateRegistry{}
+	privateRegistry.URL = registryEnabledFqdn
+	privateRegistry.IsDefault = true
+	privateRegistry.Password = registryEnabledPassword
+	privateRegistry.User = registryEnabledUsername
+	privateRegistriesAuth = append(privateRegistriesAuth, privateRegistry)
+	authClusterName, err := ranchersetup.CreateTestCluster(client, client, 1, "authregcluster", cnis[0], kubernetesVersions[0], nodesAndRoles, privateRegistriesAuth)
+	require.NoError(rt.T(), err)
+
+	// Although we wait for the cluster to be active
+	// Slow down to prevent a running condition giving 
+	// the second cluster workloads/jobs time to finish pulling images
+	time.Sleep(1 * time.Minute)
```

Comment:
I believe instead of hardcoding sleep here, another poll might expand the validation here, is there a way to poll and check this through API?

Reply:
We should never have sleeps; if the polling is not enough from the cluster, then another poll should be implemented rather than having a sleep.

Reply:
I can check if all the pods are active but that would just invalidate the tests? 
Wait until is active otherwise fail in the poll? 
Then check if is active again in the test?

Reply:
well I think that makes sense right? It's part of the test validations, so check if it's active then check the workloads/pods which ever makes sense.